libs/api/domains/notifications/src/lib/notificationsList.resolver.ts

```diff
@@ -0,0 +1,109 @@
+import type { LogoUrl, OrganizationLogoDataLoader } from '@island.is/cms'
+import { NotificationsService } from './notifications.service'
+import {
+  NotificationSender,
+  NotificationsInput,
+  NotificationsResponse,
+} from './notifications.model'
+import type { Locale } from '@island.is/shared/types'
+import { LOGGER_PROVIDER, type Logger } from '@island.is/logging'
+import { Loader } from '@island.is/nest/dataloader'
+import { AUDIT_NAMESPACE } from './notifications.resolver'
+import { FeatureFlag, Features } from '@island.is/nest/feature-flags'
+import { DocumentsScope } from '@island.is/auth/scopes'
+
+const LOG_CATEGORY = 'notification-list-resolver'
+
+@UseGuards(IdsUserGuard)
+@Resolver(() => NotificationsResponse)
+@Audit({ namespace: AUDIT_NAMESPACE })
+@Scopes(DocumentsScope.main)
+@FeatureFlag(Features.ServicePortalNotificationsEnabled)
+export class NotificationsListResolver {
+  constructor(
+    private readonly service: NotificationsService,
+    @Inject(LOGGER_PROVIDER) private readonly logger: Logger,
+  ) {}
+
+  @Query(() => NotificationsResponse, {
+    name: 'userNotifications',
+    nullable: true,
+  })
+  @Audit()
+  async getNotifications(
+    @CurrentUser() user: User,
+    @Args('input', { type: () => NotificationsInput, nullable: true })
+    input: NotificationsInput,
+    @Args('locale', { type: () => String, nullable: true })
+    locale: Locale = 'is',
+  ): Promise<NotificationsResponse | null> {
+    let notifications: NotificationsResponse | null
+
+    try {
+      notifications = await this.service.getNotifications(locale, user, input)
+    } catch (e) {
+      this.logger.error('failed to get notifications', {
+        locale,
+        category: LOG_CATEGORY,
+        error: e,
+      })
+      throw e
+    }
+
+    return notifications
+  }
+
+  @ResolveField('unseenCount', () => Int, {
+    nullable: true,
+  })
+  async resolveNotificationsUnseenCount(
+    @Context('req') { user }: { user: User },
+  ): Promise<number | undefined> {
+    const res = await this.service.getUnseenCount(user)
+    return res?.unseenCount
+  }
+
+  @ResolveField('unreadCount', () => Int, {
+    nullable: true,
+  })
+  async resolveNotificationsUnreadCount(
+    @Context('req') { user }: { user: User },
+  ): Promise<number | undefined> {
+    const res = await this.service.getUnreadCount(user)
+    return res?.unreadCount
+  }
+}
+
+@UseGuards(IdsUserGuard)
+@Resolver(() => NotificationSender)
+export class NotificationSenderResolver {
+  @ResolveField('logoUrl', () => String, { nullable: true })
+  async resolveOrganisationLogoUrl(
+    @Loader(OrganizationLogoLoader)
+    organizationLogoLoader: OrganizationLogoDataLoader,
+    @Parent() sender: NotificationSender,
+  ): Promise<LogoUrl | undefined> {
+    try {
+      return organizationLogoLoader.load({
+        key: sender?.id ?? '',
+        field: 'kennitala',
+      })
+    } catch {
+      return undefined
+    }
```

Comment:
Do we want to eat the error here?

Reply:
I just wanted to make it clear i dont care if this fails, but that's already being done within the function so ill just return it all.